datastream-common/src/main/java/com/linkedin/datastream/common/BrooklinEnvelope.java

```diff
@@ -29,14 +30,16 @@
 
   private Map<String, String> _metadata;
 
+  private Iterable<Header> _headers;
```

Comment:
can we instead use `Headers` which is an Iterable of Header? This is what the ConsumerRecord's getHeaders() returns:

`    /**
     * The headers
     */
    public Headers headers() {
        return headers;
    }`

And headers looks like this:

`public interface Headers extends Iterable<Header>`